### PR TITLE
apache-flink 1.0.2 (new formula)

### PR DESCRIPTION
**Pull request by EronWright**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document? YES
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change? YES
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)? YES
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)? YES

---

### CODE REVIEW

File: Formula/apache-flink.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def install
+    rm_f Dir["bin/*.bat"]
+    libexec.install Dir["*"]
+    bin.write_exec_script Dir["#{libexec}/bin/{flink,pyflink2.sh,pyflink3.sh}"]
+  end
+
+  test do
+    ENV["_JAVA_OPTIONS"] = "-Djava.io.tmpdir=#{testpath} -Duser.home=#{testpath}"
+    assert_match /FINISHED/, shell_output("#{libexec}/bin/start-scala-shell.sh local <<<'env.fromElements(1,2,3).print()'", 1)
```
Comment (mikemcquaid) [inline Formula/apache-flink.rb]: You'll need to kill the `scala` process afterwards too.

Comment (mikemcquaid) [inline Formula/apache-flink.rb]: Also, should you not maybe use one of the scripts you install to `bin`?

Comment (EronWright) [inline Formula/apache-flink.rb]: @mikemcquaid I don't see any residual processes after the test run - the Flink shell doesn't leave any as far as I know.   Can you elaborate on your concern?

I'd like to use one the `bin` scripts, yes, but I didn't see how to add `start-scala-shell` to `bin` without also adding it to the system path.  I don't want to add `start-scala-shell` to the system path at this time.  Are the actions separable?

Thanks

Comment (UniqMartin) [inline Formula/apache-flink.rb]: You should be able to use `#{bin}/start-scala-shell` in place of `#{libexec}/bin/start-scala-shell.sh`. Also, `pipe_output` seems to be a better fit than `shell_output`:

```ruby
pipe_output("#{bin}/start-scala-shell local", "env.fromElements(1,2,3).print()", 1)
```

**Comment by EronWright** [comment]
@UniqMartin thanks for the advice to use pipe_output, I updated accordingly.

The various Flink shells (pyshell2, pyshell3, and start-scala-shell) won't be installed to the system path at this time, pending some discussion.     

I think all feedback is addressed.

**Comment by apjanke** [comment]
LGTM.

**Comment by UniqMartin** [comment]
Merged. Thank you for this contribution to Homebrew, @EronWright! :tada:

> The various Flink shells (pyshell2, pyshell3, and start-scala-shell) won't be installed to the system path at this time, pending some discussion.

I see. Thanks for clarifying! I didn't realize this.

**Pull request merged**